File: opcua-webservice-spring/examples/uawebservicetest/src/main/java/org/opcfoundation/uawebservicetest/testdigitaltwin/DynamicSubmodelTestType.java
package org.opcfoundation.uawebservicetest.testdigitaltwin;

import org.eclipse.milo.opcua.stack.core.types.builtin.LocalizedText;
import org.opcfoundation.webserver.digitaltwin.DigitalTwinSpace;
import org.opcfoundation.webserver.digitaltwin.submodel.DynamicSubmodelType;
import org.opcfoundation.webserver.service.message.digitaltwin.GetObjectElementListRequest;
import org.opcfoundation.webserver.service.message.digitaltwin.GetObjectElementListResponse;
import org.opcfoundation.webserver.types.digitaltwin.ObjectElementDescriptor;
import java.util.concurrent.CompletableFuture;

public class DynamicSubmodelTestType extends DynamicSubmodelType {
    public DynamicSubmodelTestType(DigitalTwinSpace space) {
        super("DynamicSubmodelTestType", new LocalizedText("DynamicSubmodelTestType"), space);
        setDescription(new LocalizedText("DynamicSubmodelTestType"));
    }

    @Override
    public CompletableFuture<GetObjectElementListResponse> onGetObjectElementList(GetObjectElementListRequest request)
    {
        GetObjectElementListResponse response = new GetObjectElementListResponse();

        for (int i=request.getOffset();i<request.getOffset()+5; ++i)
        {
            response.add(new ObjectElementDescriptor(Integer.toString(i), new LocalizedText(Integer.toString(i)), DigitalTwinSpaceTest.elementCollectionTestAType));
        }

        response.setContainsMoreData(request.getOffset() == 0);

        return CompletableFuture.completedFuture(response);
    }
}

File: opcua-webservice-node/examples/testserver/src/testdigitaltwin/DynamicSubmodelTestType.ts
import { UaLocalizedText } from "opcua-webapi-ts";
import { DigitalTwinSpace } from "opcua-webservice-node";
import { DynamicSubmodelType } from "opcua-webservice-node";
import { GetObjectElementListRequest, GetObjectElementListResponse } from "opcua-webservice-node";
import { ObjectElementDescriptor } from "opcua-webservice-node";
import { DigitalTwinSpaceTest } from "./DigitalTwinSpaceTest";

export class DynamicSubmodelTestType extends DynamicSubmodelType {
    constructor(space: DigitalTwinSpace) {
        super("DynamicSubmodelTestType", new UaLocalizedText("DynamicSubmodelTestType"), space);
        this.description = new UaLocalizedText("DynamicSubmodelTestType");
    }

    override async onGetObjectElementList(request: GetObjectElementListRequest): Promise<GetObjectElementListResponse> {
        const response = new GetObjectElementListResponse();

        for (let index = request.offset; index < request.offset + 5; ++index) {
            response.add(
                new ObjectElementDescriptor(
                    `${index}`,
                    new UaLocalizedText(`${index}`),
                    DigitalTwinSpaceTest.elementCollectionTestAType,
                ),
            );
        }

        response.containsMoreData = request.offset === 0;
        return response;
    }
}